runtime/gc_modron_standard/StandardAccessBarrier.cpp

```diff
@@ -367,36 +367,24 @@ MM_StandardAccessBarrier::jniGetPrimitiveArrayCritical(J9VMThread* vmThread, jar
 	void *data = NULL;
 	J9JavaVM *javaVM = vmThread->javaVM;
 	J9InternalVMFunctions *functions = javaVM->internalVMFunctions;
+	J9IndexableObject *arrayObject = (J9IndexableObject *)J9_JNI_UNWRAP_REFERENCE(array);
+	GC_ArrayObjectModel *indexableObjectModel = &_extensions->indexableObjectModel;
 
 	bool shouldCopy = false;
 	bool alwaysCopyInCritical = (javaVM->runtimeFlags & J9_RUNTIME_ALWAYS_COPY_JNI_CRITICAL) == J9_RUNTIME_ALWAYS_COPY_JNI_CRITICAL;
-	if (alwaysCopyInCritical) {
+	if (alwaysCopyInCritical || (!indexableObjectModel->isInlineContiguousArraylet(arrayObject))) {
```

Comment:
hm, why do we need this for Standard GC?

Reply:
Unable to remember why I added this, as it was not a part of my original changeset and would have been added separately. I have removed it locally and am testing whether this is still needed.

Reply:
The underlying issue is now fixed so this change can be removed. Marking this as resolved.